README.md

```diff
@@ -118,6 +118,18 @@ delay(by: .seconds(5), dispatchLevel: .userInteractive) {
 }
 ```
 
+#### every(_ interval:) { ... }
+Runs a given closure every interval. Dispatch queue can be set optionally (as well as QoS and allowed leeway, see definition)
+``` Swift
+var counter = 0
+let timer = every(.milliseconds(10)) {
```

Comment:
Ahh, there we have a problem. Unfortunately the naming `every` doesn't tell the user that something it being returned. According to the Swift API Design Guidelines it should be clear that something is being returned here and what it might be. `every` only tells us it has to do with time. When you first mentioned `every` I didn't think it would return something but would rather work like `delay` which just runs a closure delayed.

Either we name this to something more clear like `repeatingTimerRunningEvery(.milliseconds(10))` where it would maybe lose it's "handiness" or we could actually try to make the timer cancellation handy as well. When I think about a handy closure that repeats code several times with a given pause, I'd expect that to also handle the end. So what we might change this to is something like `every(.milliseconds(10), times: 5)`.

But then we have the `times` keyword there for which we have a closure already. So why not change the existing closure to take an extra parameter like so: `5.times(every: .milliseconds(10)) { ... }`. What do you think?

Reply:
(don't you worry, I'm not lost, I just was on vacation, I'm gonna continue with this pr in a couple of days)

Reply:
Welllll, I get your concerns about this design. On the other hand, when it comes to the real life use case, I came up with it when I tried to make an infinite interval database healthcheck for my backend server. At that point I didn't need it to stop, it returned void (still failed Swift API Design Guidelines though). So from the point of abstract usability (detached from Swift ideology) my current design is quite ok. Your design however is not so usable (I'm extremely unsure `5.times` would solve real problems — like mine, about regular healthchecks), but it satisfies the guidelines. Honestly I don't know what to do now :)

Reply:
If you don't need it to stop, you can simply use `Int.max` or `Double.infinity` as the "times" count. Also if you don't like the `Double.infinity.times(every:)` approach, why don't you try to solve the naming issue (return type not clear) and the handy cancellation issue on your own solution?

For example (that's just quick thinking), what about `timerRepeatingEvery(.milliseconds(10)).cancel(.afterRepeatingTimes(5))` and `timerRepeatingEvery(.milliseconds(10)).cancel(.never)` for infinite cases by using an enum? Just a rough idea.

Reply:
Looks neat, but I'm sure there must be a possibility of stopping the execution immediately, without waiting for execution counter cap. The closure cannot know that it's inside a timer. I think something can be done with `.cancel`, but still don't know what :) Probably something like `.cancel(.semaphore(semaphoreReference))`, but it's not so swiftish. OR it should return a struct or smth with `cancel` method.

Reply:
Breaking news: you MUST keep the DispatchSourceTimer object somewhere, otherwise timer gets cancelled with garbage collection (at the end of current scope). So you cannot just call every (discarding the result), you must do something with the timer reference (let _ would not work). Soo we have another problem now.

Reply:
Okay then what about just renaming your method to  `timerRepeatingEvery` which at least solves the most apparent problem with the naming conventions? We can add the "times" feature later if we want... that should work then, no?